x-pack/plugins/ml/public/plugin.ts

```diff
@@ -103,9 +103,14 @@ export class MlPlugin implements Plugin<MlPluginSetup, MlPluginStart> {
 
     const managementApp = registerManagementSection(pluginsSetup.management, core);
 
+    // the mlUrlGenerator should be registered even if ML is not enabled
+    // for other plugins to access ML links
+    registerUrlGenerator(pluginsSetup.share, core);
```

Comment:
This shouldn't be called before the license check, we shouldn't be registering things like this if ML isn't enabled or the license isn't at least basic.
It could go just below the home page registration block.

Reply:
Thanks Jame. This is now updated here https://github.com/elastic/kibana/pull/77832/commits/025d1143e6208884c5a823138b9528b3196b7bc1